src/payment/strategies/stripe-upe/stripe-upe-payment-strategy.ts

```diff
@@ -189,6 +230,51 @@ export default class StripeUPEPaymentStrategy implements PaymentStrategy {
         throw error;
     }
 
+    private async _processVaultedAdditionalAction(
+        error: Error,
+        methodId?: string,
+        shouldSetAsDefaultInstrument = false): Promise<InternalCheckoutSelectors | never> {
+        if (!(error instanceof RequestError)) {
+            throw error;
+        }
+
+        if (!this._stripeUPEClient || !this._stripeElements) {
+            throw new NotInitializedError(NotInitializedErrorType.PaymentNotInitialized);
+        }
+
+        if (some(error.body.errors, { code: 'three_d_secure_required' }) && methodId) {
+            const clientSecret = error.body.three_ds_result.token;
+
+            const { paymentIntent, error: stripeError } = await this._stripeUPEClient.confirmCardPayment(clientSecret);
+
+            if (stripeError) {
+                this._throwDisplayableStripeError(stripeError);
+                if (this._isCancellationError(stripeError)) {
+                    throw new PaymentMethodCancelledError();
+                }
+                throw new PaymentMethodFailedError();
+            }
+            if (!paymentIntent) {
+                throw new RequestError();
+            }
+
+            const paymentPayload = {
+                methodId,
+                paymentData: {
+                    formattedPayload: {
+                        credit_card_token: { token: paymentIntent.id },
+                        confirm: false,
+                        set_as_default_stored_instrument: shouldSetAsDefaultInstrument,
+                    },
+                },
+            };
+
+            return await this._store.dispatch(this._paymentActionCreator.submitPayment(paymentPayload));
```

Comment:
Probably need to catch the error here?

Reply:
We don't catch because if it throws an error here it won't be something we can handle.